jerry-core/ecma/base/ecma-helpers-string.cpp

```diff
@@ -1880,6 +1881,82 @@ ecma_string_substr (const ecma_string_t *string_p, /**< pointer to an ecma strin
 } /* ecma_string_substr */
 
 /**
+ * Trim leading and trailing whitespace characters from string.
+ *
+ * @return trimmed ecma string
+ */
+ecma_string_t *
+ecma_string_trim (const ecma_string_t *string_p) /**< pointer to an ecma string */
+{
+  ecma_string_t *ret_string_p;
+
+  lit_utf8_size_t buffer_size = ecma_string_get_size (string_p);
+
+  if (buffer_size > 0)
+  {
+    MEM_DEFINE_LOCAL_ARRAY (utf8_str_p, buffer_size, lit_utf8_byte_t);
+    ecma_string_to_utf8_string (string_p, utf8_str_p, (ssize_t) buffer_size);
+
+    lit_utf8_iterator_t front = lit_utf8_iterator_create (utf8_str_p, buffer_size);
+    lit_utf8_iterator_seek_bos (&front);
```

Comment:
I think, line 1901 could be removed, as iterator is at the begin of string after creation.